src/components/application_manager/src/application_data_impl.cc

```diff
@@ -582,6 +582,122 @@ void DynamicApplicationDataImpl::set_display_layout(const std::string& layout) {
   window_templates_[DEFAULT_WINDOW] = template_config;
 }
 
+/**
+ * @brief RemoveUnsupportedLayouts removes unsupported layouts from
+ * configurable_keys array, if present
+ * @param keyboard_capabilities Modified keyboard capabilities
+ * @param supported_layouts Set with all supported layouts, generated from
+ * supported_keyboard_layouts array
+ * @param customizable_layouts Set with all supported layouts, which keys may be
+ * customized (if not, number of keys for such layout will be set to zero by
+ * default)
+ */
+void RemoveUnsupportedLayouts(
+    smart_objects::SmartObject& keyboard_capabilities,
+    const std::set<mobile_apis::KeyboardLayout::eType>& supported_layouts,
+    std::set<mobile_apis::KeyboardLayout::eType>& customizable_layouts) {
```

Comment:
@ychernysheva as your function is `void`, please use return value instead of non-const reference:
```suggestion
std::set<mobile_apis::KeyboardLayout::eType> RemoveUnsupportedLayouts(
    smart_objects::SmartObject& keyboard_capabilities,
    const std::set<mobile_apis::KeyboardLayout::eType>& supported_layouts) {
```

Also, you may use a const reference in external function to bind the temp object

Reply:
Done in a9c3f0b